### PR TITLE
Preserve `class` casing when merging a `TagBuilder` into a `TagHelperOutput`

### PR DESCRIPTION
**Pull request by dougbu**
- #5313

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs [hunk 1]
```diff
@@ -110,18 +109,15 @@ public static class TagHelperOutputExtensions
...
+                    var found = tagHelperOutput.Attributes.TryGetAttribute("class", out classAttribute);
+                    Debug.Assert(found);
 
-                    if (tagHelperOutput.Attributes.TryGetAttribute("class", out classAttribute))
-                    {
-                        tagHelperOutput.Attributes.SetAttribute("class", classAttribute.Value + " " + attribute.Value);
-                    }
-                    else
-                    {
-                        tagHelperOutput.Attributes.Add("class", attribute.Value);
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs]: Note this case was impossible.

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs [hunk 2]
```diff
@@ -110,18 +109,15 @@ public static class TagHelperOutputExtensions
...
-                    }
+                    tagHelperOutput.Attributes.SetAttribute(
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs]: Also update the ValueStyle so the quotes match.

Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs]: Done here and in a few tag helpers.

**Comment by dougbu** [comment]
:up::date:

File: test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CaseSensitiveTagHelperAttributeComparer.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    public class CaseSensitiveTagHelperAttributeComparer : IEqualityComparer<TagHelperAttribute>
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CaseSensitiveTagHelperAttributeComparer.cs]: Probably worth having this in https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNetCore.Razor.Test.Sources
and nuking [the one in Razor](https://github.com/aspnet/Razor/blob/e14f4b095d0bd4c5a5ae079f817f9a07217907af/test/Microsoft.AspNetCore.Razor.Runtime.Test/Runtime/TagHelpers/CaseSensitiveTagHelperAttributeComparer.cs).

Could be separate, just a thought 😄

**Comment by dougbu** [comment]
27e4822

File: test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CaseSensitiveTagHelperAttributeComparer.cs (hunk 3 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CaseSensitiveTagHelperAttributeComparer.cs]: #5323

**Pull request closed without merging**